.github/workflows/typespec-validation-ci.yml

```diff
@@ -0,0 +1,58 @@
+# TODO: Refactor to eliminate duplication with suppressions-ci.yaml
+
+name: TypeSpec Validation - CI
```

Comment:
How about 

`Test TypeSpec Validation - CI`

Reply:
I used the same name as the DevOps pipeline it is replacing.  But I agree this is a good time to define a better naming scheme for GH actions (at least within the azure-rest-api-specs repo).

I'm planning to move checks `TypeSpec Validation` and `TypeSpec Validation - All` to GH Actions soon, and I'd like to preserve `these` names since spec authors and team members are already familiar.

How about `Test - TypeSpec Validation` or `Unit Test - TypeSpec Validation`?  The latter is more explicit, but also maybe too explicit (e.g. what even is a "unit" test, the `npm` command is just "test").